src/csharp/Microsoft.Spark/ML/Feature/NGram.cs

```diff
@@ -0,0 +1,111 @@
+        /// </summary>
+        public NGram() : base(s_NGramClassName)
+        {
+        }
+
+        /// <summary>
+        /// Create a <see cref="NGram"/> with a UID that is used to give the
+        /// <see cref="NGram"/> a unique ID.
+        /// </summary>
+        /// <param name="uid">An immutable unique ID for the object and its derivatives.</param>
+        public NGram(string uid) : base(s_NGramClassName, uid)
+        {
+        }
+
+        internal NGram(JvmObjectReference jvmObject) : base(jvmObject)
+        {
+        }
+
+        JvmObjectReference IJvmObjectReferenceProvider.Reference => _jvmObject;
+
+        /// <summary>
+        /// Gets the column that the <see cref="NGram"/> should read from.
+        /// </summary>
+        /// <returns>string, input column</returns>
+        public string GetInputCol() => (string)_jvmObject.Invoke("getInputCol");
+
+        /// <summary>
+        /// Sets the column that the <see cref="NGram"/> should read from.
+        /// </summary>
+        /// <param name="value">The name of the column to as the source</param>
+        /// <returns>New <see cref="NGram"/> object</returns>
+        public NGram SetInputCol(string value) => WrapAsNGram(_jvmObject.Invoke("setInputCol", value));
+
+        /// <summary>
+        /// The <see cref="NGram"/> will create a new column in the DataFrame, this is the name of the new
+        /// column.
+        /// </summary>
+        /// <returns>string, the output column</returns>
+        public string GetOutputCol() => (string)_jvmObject.Invoke("getOutputCol");
+
+        /// <summary>
+        /// The <see cref="NGram"/> will create a new column in the DataFrame, this is the
+        /// name of the new column.
+        /// </summary>
+        /// <param name="value">The name of the new column</param>
+        /// <returns>New <see cref="NGram"/> object</returns>
+        public NGram SetOutputCol(string value) =>
+            WrapAsNGram(_jvmObject.Invoke("setOutputCol", value));
+
+        /// <summary>
+        /// Gets N value for <see cref="NGram"/>.
+        /// </summary>
+        /// <returns>int, N value</returns>
+        public int GetN() => (int)_jvmObject.Invoke("getN");
+
+        /// <summary>
+        /// Sets N value for <see cref="NGram"/>.
+        /// </summary>
+        /// <param name="value">N value</param>
+        /// <returns>New <see cref="NGram"/> object</returns>
+        public NGram SetN(int value) => WrapAsNGram(_jvmObject.Invoke("setN", value));
+
+        /// <summary>
+        /// Executes the <see cref="NGram"/> and transforms the DataFrame to include the new column.
+        /// </summary>
+        /// <param name="source">The DataFrame to transform</param>
+        /// <returns>
+        /// New <see cref="DataFrame"/> object with the source <see cref="DataFrame"/> transformed.
+        /// </returns>
+        public DataFrame Transform(DataFrame source) =>
```

Comment:
Could you also add TransformSchema so it is consistent with the other transformers?